content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,408 @@
+- [ ] If appropriate, a mechanism to inject secrets stored in third-party storage
+  is deployed and available.
+- [ ] Service account tokens are not mounted in pods that don't require them.
+- [ ] [Bound service account token volume](/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume)
+  is in-use instead of non-expiring tokens.
+
+Secrets required for pods should be stored within Kubernetes Secrets as opposed
+to alternatives such as ConfigMap. Secret resources stored within etcd should
+be [encrypted at rest](/docs/tasks/administer-cluster/encrypt-data/).
+
+Pods needing secrets should have these automatically mounted through volumes,
+preferably stored in memory like with the [`emptyDir.medium` option](/docs/concepts/storage/volumes/#emptydir).
+Mechanism can be used to also inject secrets from third-party storages as
+volume, like the [Secrets Store CSI Driver](https://secrets-store-csi-driver.sigs.k8s.io/).
+This should be done preferentially as compared to providing the pods service
+account RBAC access to secrets. This would allow adding secrets into the pod as
+environment variables or files. Please note that the environment variable method
+might be more prone to leakage due to crash dumps in logs and the
+non-confidential nature of environment variable in Linux, as opposed to the
+permission mechanism on files.
+
+Service account tokens should not be mounted into pods that do not require them. This can be configured by setting
+[`automountServiceAccountToken`](/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server)
+to `false` either within the service account to apply throughout the namespace
+or specifically for a pod. For Kubernetes v1.22 and above, use
+[Bound Service Accounts](/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume)
+for time-bound service account credentials.
+
+## Images
+
+- [ ] Minimize unnecessary content in container images.
+- [ ] Container images are configured to be run as unprivileged user.
+- [ ] References to container images are made by sha256 digests (rather than
+tags) or the provenance of the image is validated by verifying the image's
+digital signature at deploy time [via admission control](/docs/tasks/administer-cluster/verify-signed-images/#verifying-image-signatures-with-admission-controller).
+- [ ] Container images are regularly scanned during creation and in deployment, and
+  known vulnerable software is patched.
+
+Container image should contain the bare minimum to run the program they
+package. Preferably, only the program and its dependencies, building the image
+from the minimal possible base. In particular, image used in production should not
+contain shells or debugging utilities, as an
+[ephemeral debug container](/docs/tasks/debug/debug-application/debug-running-pod/#ephemeral-container)
+can be used for troubleshooting.
```

Comment:
We should explain and recommend multistage builds as _one_ technique that helps minimize what gets included.